src/ao/oc_ao_options.lsl

```diff
@@ -69,749 +70,315 @@ integer CMD_OWNER = 500;
-    }
-    if(AOSit) // Apply white on/off setting to sit prim
-    {   
-        llSetLinkColor(3, AOoncolor , ALL_SIDES); 
-    }
-    else
-    {
-        llSetLinkColor(3, AOoffcolor , ALL_SIDES);  
-    }
+DoStatus() {
+    vector color;
+    if (g_iAOPower) color = g_vAOoncolor;
+    else color = g_vAOoffcolor;
+    llSetLinkColor(llListFindList(g_lButtons,["Power"]), color, ALL_SIDES);
+    if (g_iAOSit) color = g_vAOoncolor;
+    else color = g_vAOoffcolor;
+    llSetLinkColor(llListFindList(g_lButtons,["SitAny"]), color, ALL_SIDES);
 }
 
-DoReset()
-{   // -- Reset the entire HUD back to default
+MainMenu(key id) {
+    g_sCurrentMenu = g_sHudMenu;
+    string text = "\nThis menu sets your HUD options.\n";
+    text += "[Horizontal] sets the button layout to Horizontal.\n";
+    text += "[Vertical] sets the button layout to Vertical.\n";
+    text += "[Order] opens the sub menus to reorder the buttons.\n";
+    text += "[Dark] or [Light] chooses a dark or light button.\n";
 
-    Layout = 1;
-    SPosition = 69; // -- Don't we just love that position? *winks*
-    tintable = FALSE;
-    Hidden = FALSE;
-    AOLock = FALSE;
-    AOPower = TRUE;
-    AOSit = FALSE;
-    DoTextures("White");
-    llSleep(1.5);
-    primOrder = [0,1,2,3,4];
-    DoHide();
-    llSleep(1.0);
-    DefinePosition();
-    DoStatus();
-    llSleep(1.5); // -- We want the position to be set before reset
-    llOwnerSay("Finalizing HUD Reset... please wait a few seconds so all menus have time to initialize.");
-    llResetScript();
-}    
-// End HUD Options    
+    list buttons = ["Horizontal","Vertical","Order"];
+    buttons += g_lStyles;
+    g_kMenuID = Dialog(id, text, buttons, [UPMENU], 0);
+}
 
-// Start standard 
-default
-{
-    changed(integer c)
-    {
-        if (c & CHANGED_OWNER) // Nice way to do this and not break everything in here
-        {
-            //DoTextures("White");
-           // llGiveInventory(llGetOwner(),"OpenCollar SubAO Help Image");
-            llResetScript();
-        }
-        else if (c & CHANGED_COLOR)
-        {
-            DetermineColors(); // -- If we change color because of tint, we need to set the new AOoffcolor!
+OrderMenu(key id) {
+    g_sCurrentMenu = g_sOrderMenu;
+    string text = "This is the order menu, simply select the\n";
+    text += "button which you want to re-order.\n\n";
+    integer i;
+    list buttons;
+    for (i=2;i<llGetListLength(g_lPrimOrder);++i) {
+        integer pos = llList2Integer(g_lPrimOrder,i);
+        buttons += llList2List(g_lButtons,pos,pos);
+    }
+    g_kMenuID = Dialog(id, text, buttons, ["Reset",UPMENU], 0);
+}
+
+
+default {
+    changed(integer change) {
+        if (change & CHANGED_OWNER) llResetScript();
+        else if (change & CHANGED_LINK) llResetScript();
+        else if (change & CHANGED_COLOR) {
+            DetermineColors(); // -- If we change color because of tint, we need to set the new g_vAOoffcolor!
             DoStatus();
```

Comment:
this causes a loop, color changed, determineColors(), Do Status() color changes again... until all is black